plugin-name/phpstan.neon

```diff
@@ -1,64 +1,53 @@
-        # https://core.trac.wordpress.org/ticket/43304
-        - '/^Parameter #2 \$deprecated of function load_plugin_textdomain expects string, false given\.$/'
-        # WP-CLI accepts a class as callable
-        - '/^Parameter #2 \$callable of static method WP_CLI::add_command\(\) expects callable\(\): mixed, \S+ given\.$/'
-        # Stubs for WP-CLI are not available
-        - '#on an unknown class WP_CLI\.#'
-        # void as typehint is available only on PHP 7.1+
-        - '#has no return typehint specified\.#'
-        # CMB2 is not detected by PHPStan
-        - '#cmb2#'
-        - '#Cmb2#'
-        # Freemius is not detect by PHPStan
-        - '#fs_dynamic_init#'
-        # Uninstall is not detect by PHPStan
-        - '#pn_uninstall#'
-        # Other method to silence that PHPStan cannot detect
-        - '#Call to an undefined method object::is_#'
-        - "#Cannot access offset 'id' on object.#"
-        - '#Access to an undefined property object::\$query.#'
-        - '#invalid type array<WP_Term>\|int supplied for foreach#'
-        - '#Function is_amp_endpoint not found.#'
+        # TODO Add file names!!
+
+        # Void as typehints are available only on PHP 7.1+
+        - '#has no return typehint specified\.$#'
+        # In wp_count_posts(): $counts = (object) $counts;
+        - '#^Access to an undefined property object::\$pending\.$#'
+        # AMP plugin stubs can be also generated
+        - '#^Function is_amp_endpoint not found\.$#'
+        # PHPStan is better safe than sorry!
+        # https://github.com/phpstan/phpstan/issues/1060#issuecomment-667675767
+        - '#^Argument of an invalid type object supplied for foreach, only iterables are supported\.$#'
+        # https://github.com/WPBP/WordPress-Plugin-Boilerplate-Powered/issues/162
+        - '#^Parameter \#2 \$object_id of function cmb2_metabox_form expects int, string given\.$#'
+        # https://github.com/Freemius/wordpress-sdk/pull/439
+        - '#^Parameter \#1 \$module of function fs_dynamic_init expects array<string, string>, array<string, array<string, string>\|bool\|string> given\.$#'
+        # TODO Any can POST an array in a $_POST item, May I send a PR?
+        - '#^Parameter \#1 \$str of function sanitize_text_field expects string, array<string>\|string given\.$#'
+# Not used in the boilerplate.
+#        # Uses func_get_args()
+#        - '#^Function apply_filters invoked with [34567] parameters, 2 required\.$#'
+#        # https://core.trac.wordpress.org/ticket/43304
+#        - '/^Parameter #2 \$deprecated of function load_plugin_textdomain expects string, false given\.$/'
+#        # WP-CLI accepts a class as callable
+#        - '/^Parameter #2 \$callable of static method WP_CLI::add_command\(\) expects callable\(\): mixed, \S+ given\.$/'
+#        # WP-CLI stubs can be generated and included in "scanFiles:"
+#        # https://github.com/szepeviktor/phpstan-wordpress/blob/master/stub-generators/wp-cli-generate-stubs.sh
```

Comment:
So they are not provided by default with the package and require a manual generation?
If it is yes I need to think how to automatize this step.

Reply:
You are welcome to contribute another repository to the php-stubs organitazion!
https://github.com/php-stubs

Currently it has no WP-CLI or AMP plugin stubs.

Reply:
I am happy to help with that, if we can have a wp-cli stub package it will be awesome.
For AMP looking how much is becoming popular is important too.

Reply:
Done on Slack.